### ISSUE TITLE
Links to same route to List view with different filters

### ISSUE THREAD

**Issue by wadjeroudi** [marmelab/admin-on-rest#342]
I've a custom menu and I've two links for the same route "items" but with different filters : 
```js
            <ListItem
                containerElement={<Link to={{ pathname: 'items', query: { filter: JSON.stringify({ type: 'no_packaging' }) } }} />}
                leftIcon={<ItemIcon />}
                primaryText="Items"
            />
            <ListItem
                containerElement={<Link to={{ pathname: 'items', query: { filter: JSON.stringify({ type: 'box' }) } }} />}
                leftIcon={<BoxIcon />}
                primaryText="Boxes"
            />
```

If I come from a different route than "items", everything works, the list component add the missing parameters, sort, page...etc
But If I switch between those two links on the same route, it doesn't reapply the filters, and the parameters are missing.

I tried to find the problem, but I didn't succeed.

**Comment by wadjeroudi** [comment]
I think that the function changeListParams should be called after new props at this line:
https://github.com/marmelab/admin-on-rest/blob/master/src/mui/list/List.js#L79
After 
```js
this.updateData(Object.keys(nextProps.query).length > 0 ? nextProps.query : nextProps.params);
```
@fzaninotto Tell me if I'm correct, and will make a PR.

**Comment by wadjeroudi** [comment]
#133 same issue.

**Comment by wadjeroudi** [comment]
I didn't succeed to fix the problem in the List component.
The problem seems simple but the implementation of filterValues is complicated (filters come from the form and the query, so to decide when to reset the filters is difficult).

So I decided to add custom routes that duplicates my list component (it's a more easy solution because I had to show different columns depending on the filters used). So now I've two List component for the same resource but with different columns and default filters.

```js
import React from 'react';
import { IndexRoute, Redirect, Route } from 'react-router';
import { createRoutesFromReactChildren } from 'react-router/lib//RouteUtils';
import pure from 'recompose/pure';
import { ItemsBoxesList } from './items';

const DuplicateRoute = () => <div>&lt;DuplicateRoute&gt; elements are for configuration only and should not be rendered</div>;

DuplicateRoute.createRouteFromReactElement = (element, parentRoute) => {
    const { path, component, resource, hasCreate, hasDelete, hasEdit, hasShow } = element.props;
    // dynamically add crud route
    const duplicateRoute = createRoutesFromReactChildren(
        <Route path={path}>
            <IndexRoute component={component} />
            <Redirect from="create" to={`/${resource}/create`} />
            <Redirect from=":id/show" to={`/${resource}/:id/show`} />
            <Redirect from=":id/edit" to={`/${resource}/:id/edit`} />
            <Redirect from=":id/delete" to={`/${resource}/:id/delete`} />
        </Route>,
        parentRoute,
    )[0];
    // higher-order component to pass myprop as resource to components
    duplicateRoute.component = pure(({ children }) => (
        <div>
            {React.Children.map(children, child => React.cloneElement(child, { resource, hasList: true, hasCreate: !!hasCreate, hasShow: !!hasShow, hasEdit: !!hasEdit, hasDelete: !!hasDelete }))}
        </div>
    ));
    return duplicateRoute;
};

export default () => (
    <DuplicateRoute path="/items-boxes" component={ItemsBoxesList} resource="items" hasShow />
);
```

I don't close the ticket, may be some day it will be fixed.

**Comment by fzaninotto** [comment]
This is fixed by #541 (currently in `next` branch, not yet released).
